experimental/learn_descriptors/four_seasons_parser.hh

```diff
@@ -1,144 +1,56 @@
 #pragma once
 
-#include <stdio.h>
-
 #include <cstddef>
 #include <filesystem>
+#include <memory>
 #include <opencv2/opencv.hpp>
 #include <string>
 #include <vector>
 
 #include "common/liegroups/se3.hh"
-#include "experimental/learn_descriptors/image_point.hh"
-#include "nmea/message/gga.hpp"
-#include "nmea/message/rmc.hpp"
-#include "nmea/sentence.hpp"
+#include "experimental/learn_descriptors/camera_calibration.hh"
+#include "experimental/learn_descriptors/four_seasons_transforms.hh"
+#include "experimental/learn_descriptors/image_point_four_seasons.hh"
 
 namespace robot::experimental::learn_descriptors {
-class FourSeasonsParser {
+class FourSeasonsParser : std::enable_shared_from_this<FourSeasonsParser> {
```

Comment:
is `std::enable_shared_from_this<>` still needed after you made the `SharedTransforms`  and `CameraCalibrationFisheye` structs shared?